File: saleup-spring/src/main/java/com/anaselrayan/springcashiero/features/products/request/ProductRequest.java
package com.anaselrayan.springcashiero.features.products.request;

import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import lombok.Data;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;

@Data
public class ProductRequest {

    private Long productId;

    private Long productCategoryId;

    private Long unitOfMeasureId;

    private Long productStatusId;

    private Long productBrandId;

    private List<MultipartFile> imageFiles;

    @NotNull(message = "product name can't be null")
    @NotEmpty(message = "product name can't be empty")
    private String productName;

    @NotNull(message = "product name can't be null")
    @NotEmpty(message = "product name can't be empty")
    private String barcode;

    private String description;

    private Integer lowStockPoint;

    private Integer quantity;

    @NotNull(message = "product cost can't be null")
    private Double costPrice;

    @NotNull(message = "product price can't be null")
    private Double sellingPrice;

}

File: saleup-ng/src/app/modules/products/models/product-request.ts
import { DiscountType } from "./product.model";

export class ProductRequest {
  productId?: number;
  productCategoryId?: number;
  unitOfMeasureId?: number;
  productStatusId?: number;
  productBrandId?: number;
  imageFiles?: File[];
  productName?: string;
  barcode?: string;
  description?: string;
  lowStockPoint?: number;
  quantity?: number;
  costPrice?: number;
  sellingPrice?: number
}

export class ProductBrandRequest {
  productBrandId?: number;
  name?: string;
  description?: string;
}

export class ProductCategoryRequest {
  productCategoryId?: number;
  name?: string;
  description?: string;
}

export class ProductUnitRequest {
  unitOfMeasureId?: number;
  code?: string;
  name?: string;
}

export class ProductDiscountRequest {
  productId!: number;
  discountActive!: boolean;
  discountType!: DiscountType;
  discountAmount!: number;
  discountStartDate!: Date;
  discountEndDate!: Date;
}